### DISCUSSION TITLE
clickable parent item in hierarchical menu

### DISCUSSION THREAD

**Discussion by artmedia** [oqtane/oqtane.framework#1092] [Ideas]
Hello,
hierarchical menu is a nice new feature in recent updates.
I noticed that parent menu item is clickable
so when I click on "ParentPage" it opens  http://localhost:44357/ParentPage (which is an empty page).
Clicking on on parent menu item should only open child sub-menu items.
Is it possible to add an option to "disable" click on parent menu item?

**Comment by hishamco** [comment]
Please steps to reproduce, so I can fix the issue?

**Reply by sbwalker** [reply]
@artmedia I could be wrong but I think what you are really asking for is a way to disable the navigation to the parent page - but not disable the click, as it is the click event which allows you to view the child menu options.

**Reply by artmedia** [reply]
@hishamco to reproduce it, just create a page "ParentPage" al root level and two child pages "ChildPage1, ChildPage2" under ParentPage.
When you want to go to a Child page you need 3 clicks, first click makes you go to http://localhost:44357/ParentPage,
then the second click opens sub-menu, and the third click makes you go to child page.
The first click is not needed.

**Reply by artmedia** [reply]
@sbwalker yes, you are right

**Reply by sbwalker** [reply]
@artmedia so I think the tricky aspect of this item is that in a desktop browser scenario you would want the parent page link to be active and clickable for navigation, and when the user hovers over the link they would expect the child menu to be displayed. However this behavior does not work in a mobile scenario as there is no hover action available. And I think most sites nowadays would like a menu which behaves well in both desktop and mobile scenarios. So one option is to allow the parent page to still be clickable for navigation, and only utilize the down arrow icon as the element which can be clicked to display the child menu. This option would still be intuitive to users and preserves the ability to navigate to parent pages without any restriction ( which is a valid scenario in many sites ).

**Comment by hishamco** [comment]
I just notice the behavior, I think there should be a distinction between child pages and submenu items, coz sometimes I need a sub menu items without creating a parent pagee. IMHO we should add an option for the navigation if the parent page doesn't have a URL the menu should be open

**Reply by sbwalker** [reply]
It sounds like you want the Disabled option from DNN:

http://www.ibew.org/Portals/31/DotNetNuke_7_0_6_User_Manual.pdf#page640

**Reply by hishamco** [reply]
Before I dig into the docs, I could say yes, if sounds looks good, I will submit a PR

**Reply by hishamco** [reply]
Seems the link is broken ;( anyhow adding a `Disabled` property may solve the issue

**Reply by artmedia** [reply]
@hishamco @sbwalker 
Can you add Disabled option (like DNN) in Page properties?
tnx

**Reply by sbwalker** [reply]
@artmedia you are referring to a Page "Navigation" property:

Disabled:
The page will still be visible in the menu; however no action is taken when a user clicks on the link in the menu. An example of when you might disable a link is when you have a parent page that doesn't have any content, but still allows users to navigate to its children.

Enabled (default):
Select this option to enable the page link in the menu for navigation.

Since this requires a database change, it will need to use the new migrations approach which is still being finalized for the 2.1 release

**Reply by artmedia** [reply]
Great!

**Reply by curelom** [reply]
Looks like this didn't make it into version 2.1

**Reply by hishamco** [reply]
Nope, I can have a look to this to be a part of `2.2.0` if not `3.0.0`

**Reply by sbwalker** [reply]
The scope of 2.1.0 was very large with all of the changes required to support multiple databases. Adding a Navigation property should be fairly straightforward so it may even fit in a 2.1.1 release.

**Reply by sbwalker** [reply]
this is in #1465 for 2.1.1
